io/disk/softwareraid.py

```diff
@@ -40,116 +39,58 @@ class SoftwareRaid(Test):
-            cmd += " --force"
-        self.check_pass(cmd, "Failed to create a MD device")
-        self.mdadm_detail()
-
-    def extensivetest(self):
-        """
-        Extensive software raid options are run viz create, delete, assemble,
-        create spares, remove and add drives
-        """
-        cmd = "mdadm --fail %s %s" % (self.raid, self.remadd)
-        self.check_pass(cmd, "Unable to fail a drive from MD device")
-        self.mdadm_detail()
-        cmd = "mdadm --manage %s --remove %s" % (self.raid, self.remadd)
-        self.check_pass(cmd, "Failed to remove a drive from MD device")
-        self.mdadm_detail()
-        cmd = "mdadm --manage %s --add %s" % (self.raid, self.remadd)
-        self.check_pass(cmd, "Failed to add back the drive to MD device")
-        self.mdadm_detail()
-        cmd = "mdadm --manage %s --stop" % self.raid
-        self.check_pass(cmd, "Failed to stop/remove the MD device")
-        cmd = "mdadm --assemble %s %s %s" \
-              % (self.raid, self.disk, self.sparedisk)
-        self.check_pass(cmd, "Failed to assemble back the MD device")
-        while self.is_mdadm_recovering():
-            time.sleep(30)
-        process.system(cmd, ignore_status=True, shell=True)
-        self.mdadm_detail()
-
-    def mdadm_detail(self):
-        """
-        Function to print the details of mdadm array
-        """
-        cmd = "mdadm --detail %s" % self.raid
-        output = process.run(cmd, ignore_status=True, shell=True)
-        if output.exit_status != 0:
-            self.fail("Failed to display MD device details")
-        return output.stdout.decode("utf-8").splitlines()
-
-    def is_mdadm_recovering(self):
-        """
-        Function to check if the array is recovering or not
-        """
-        for line in self.mdadm_detail():
-            if 'State' in line and 'recovering' in line:
-                return True
-        return False
-
-    def check_pass(self, cmd, errmsg):
-        """
-        Function to check if the cmd is successful or not, if not display
-        appropriate message
-        """
-        if process.system(cmd, ignore_status=True, shell=True) != 0:
-            self.fail(errmsg)
+        disk = self.params.get('disks', default='').strip(" ")
+        if not disk:
+            self.cancel('No disks given')
+        required_disks = self.params.get('required_disks', default=1)
+        disk = disk.split()
+        raidlevel = str(self.params.get('raid', default='0'))
+        if len(disk) < required_disks:
+            self.cancel("Minimum %d disks required for %s" % (required_disks,
+                                                              raidlevel))
+        spare = self.params.get('spare_disks', default='').strip(" ")
+        if spare:
+            spare = spare.split()
+        raid = self.params.get('raidname', default='/dev/md/mdsraid')
+        metadata = str(self.params.get('metadata', default='1.2'))
+        self.remadd = ''
```

Comment:
self.remadd can be like "operational_disk". as we are performing operations like remove, add, stop, assemble etc.

or something more meaning conveying name.

Reply:
The operation we are doing with it is remove and add, so using remadd.

Reply:
ok, then can we make it like rem_add_disk/remove_add_disk?

Reply:
done